<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Footnotes

## Exercise 1

In [2]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

I pressed Kernel on the top left and the Interrupt Kernel button in the drop down menu

## Exercise 2

In [3]:
print = 'PRINT'

In [4]:
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello World')

Hello World


## Exercise 3

In [52]:
squares1 = [i**2 for i in range(10)]

In [53]:
print(squares1)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [54]:
squares2 = list()
for i in range(10):
    squares2.append(i**2)

In [55]:
print(squares2)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [56]:
%%timeit
squares1
squares2

70 ns ± 8.81 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)


it takes on average 70 nanoseconds to execute both lines of code with a standard deviation of 8.81 nanoseconds. Each code was executed 10,000,000 times. The average time was taken with the best 7 runs